doc/man7/openssl-quic.pod

```diff
@@ -0,0 +1,679 @@
+
+=item L<BIO_get_rpoll_descriptor(3)>, L<BIO_get_wpoll_descriptor(3)>
+
+This is a new BIO API which allows a BIO to expose a poll descriptor. This API
+is used to implement the corresponding SSL APIs L<SSL_get_rpoll_descriptor(3)>
+and L<SSL_get_wpoll_descriptor(3)>.
+
+=item L<BIO_sendmmsg(3)>, L<BIO_recvmmsg(3)>
+
+This is a new BIO API which can be implemented by BIOs which implement datagram
+semantics. It is implemented by L<BIO_s_datagram(3)> and L<BIO_s_dgram_pair(3)>.
+It is used by the QUIC implementation to send and receive UDP datagrams.
+
+=item L<BIO_dgram_set_no_trunc(3)>, L<BIO_dgram_get_no_trunc(3)>
+
+By default, L<BIO_s_dgram_pair(3)> has semantics comparable to those of Berkeley
+sockets being used with datagram semantics. This allows an alternative mode
+to be enabled in which datagrams will not be silently truncated if they are
+too large.
+
+=item L<BIO_dgram_set_caps(3)>, L<BIO_dgram_get_caps(3)>
+
+These functions are used to allow the user of one end of a
+L<BIO_s_dgram_pair(3)> to indicate its capabilities to the other end of a
+L<BIO_s_dgram_pair(3)>. In particular, this allows an application to inform the
+QUIC implementation of whether it is prepared to handle local and/or peer
+addresses in transmitted datagrams and to provide the applicable information in
+received datagrams.
+
+=item L<BIO_dgram_get_local_addr_cap(3)>, L<BIO_dgram_set_local_addr_enable(3)>,
+L<BIO_dgram_get_local_addr_enable(3)>
+
+Local addressing support refers to the ability of a BIO with datagram semantics
+to allow a source address to be specified on transmission and to report the
+destination address on reception. These functions can be used to determine if a
+BIO can support local addressing and to enable local addressing support if it
+can.
+
+=item L<BIO_err_is_non_fatal(3)>
+
+This is used to determine if an error while calling L<BIO_sendmmsg(3)> or
+L<BIO_recvmmsg(3)> is ephemeral in nature, such as "would block" errors.
+
+=back
+
+=head1 THREAD ASSISTED MODE
+
+The optional thread assisted mode can be used with
+L<OSSL_QUIC_client_thread_method(3)>. In this mode, a background thread is
+created automatically. The OpenSSL QUIC implementation then takes responsibility
+for ensuring that timeout events are handled on a timely basis even if no SSL
+I/O function such as L<SSL_read(3)> or L<SSL_write(3)> is called by the
+application for a long time.
+
+All necessary locking is handled automatically internally, but the thread safety
+guarantees for the public SSL API are unchanged. Therefore, an application must
+still do its own locking if it wishes to make concurrent use of the public SSL
+APIs.
+
+Because this method relies on threads, it is not available on platforms where
+threading support is not available or not supported by OpenSSL. However, it
+does provide the simplest mode of usage for an application.
+
+The implementation may or may not use a common thread or thread pool to service
+multiple SSL objects in the same B<SSL_CTX>.
```

Comment:
Currently the implementation seems to only handle a single object

Reply:
We currently create one thread per thread-assisted connection. But we reserve the right to optimise this in the future.